### PR TITLE
implemented basic throttling functionality

### PR DESCRIPTION
**Pull request by skaes**
Intended to be used for slowing down fast publishers.

### CODE REVIEW

File: examples/throttling.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# simple.rb
```
Comment (boosty) [inline examples/throttling.rb]: Should be `throttling.rb`

File: examples/throttling.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# this example shows you a very basic message/handler setup
+#
+# ! check the examples/README.rdoc for information on starting your redis/rabbit !
+#
+# start it with ruby simple.rb
```
Comment (boosty) [inline examples/throttling.rb]: Again, should be `throttling.rb`

File: examples/throttling.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# simple.rb
+# this example shows you a very basic message/handler setup
```
Comment (boosty) [inline examples/throttling.rb]: I think this description should mention the throttling behaviour we want to test.

File: lib/beetle/client.rb [hunk 4]
```diff
@@ -258,6 +258,25 @@ def resume_listening(*queues)
...
+    # throttle publishing of messages based on queue lengths.
+    #
+    #    client.throttle(:foo => 10000, :bar => 10_000)
+    #
+    # throttle publisher to 1 message per second as long as queue foo has more
+    # than 10000 entries or queue bar has more than 100000 entries across all
+    # servers. Queue lenghts are periodically determined during publishing. You
+    # only want to use this feature if you plan to publish a huge amount of
+    # messages to slow consumers so as to not overload the broker or the redis
+    # dedpuplication store.
```
Comment (boosty) [inline lib/beetle/client.rb]: `s/dedpuplication/deduplication`

File: lib/beetle/publisher.rb [hunk 5]
```diff
@@ -135,6 +151,16 @@ def rpc(message_name, data, opts={}) #:nodoc:
...
     end
 
+    def throttle(queue_options)
+      @throttling_options = queue_options
+    end
+
+    def throttle!
+      return unless throttling?
+      refresh_throttling!
+      sleep 1 if throttled?
```
Comment (boosty) [inline lib/beetle/publisher.rb]: 1 second might be ok for a background process (a worker or cronjob), but seems may too long for an app or api process. I suggest to make this a config option. 

Or do we suggest to better not use the throttling in app and api processes at all?

File: examples/throttling.rb (hunk 1 above)
Comment (skaes) [reply examples/throttling.rb]: fixed

File: examples/throttling.rb (hunk 2 above)
Comment (skaes) [reply examples/throttling.rb]: fixed

File: examples/throttling.rb (hunk 3 above)
Comment (skaes) [reply examples/throttling.rb]: fixed

File: lib/beetle/client.rb (hunk 4 above)
Comment (skaes) [reply lib/beetle/client.rb]: fixed

File: lib/beetle/publisher.rb (hunk 5 above)
Comment (skaes) [reply lib/beetle/publisher.rb]: We only want to throttle background workers/cronjobs.

**Pull request merged**